### DISCUSSION TITLE
Need Help : Not injecting span_id and trace_id in log

### DISCUSSION THREAD

**Discussion by Yogeshsuch** [open-telemetry/opentelemetry-java-instrumentation#11947] [Q&A] [answered]
I have created a simple Maven project using Intellij, and I have written a code for accepting an HTTP request. We are using manual instrumentation. 

I have followed the following documentation : https://help.sumologic.com/docs/apm/traces/get-started-transaction-tracing/opentelemetry-instrumentation/java/traceid-spanid-injection-into-logs-configuration/

I expect to see the actual trace id and span id but not able to see it in the logs.

Can someone please help where I need to change

**Comment by jaydeluca** [comment]
> I have created a simple Maven project using Intellij

Are you able to upload this code to github for review? it might be easier to assist in troubleshooting with more context

**Reply by Yogeshsuch** [reply]
Not possible to upload code I will share snippets and for logback configuration I refer https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/logback/logback-appender-1.0/library

**Reply by Yogeshsuch** [reply]
ContextPropagators propagators = ContextPropagators.create(W3CTraceContextPropagator.getInstance());

//Initialize tracerProvider
SdkTracerProvider tracerProvider = SdkTracerProvider.builder()
        .setResource(resource)
        .setSampler(headerBasedSampler)
        .addSpanProcessor(SimpleSpanProcessor.create(spanExporter))
        .build();
        
//Set opentelemetySDK globally
    OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
            .setTracerProvider(tracerProvider)
            .setPropagators(propagators)
            .buildAndRegisterGlobal();

**Reply by laurit** [reply]
Please provide a minimal application that reproduces your issue along with any instructions necessary.

**Reply by Yogeshsuch** [reply]
[otel_demo.zip](https://github.com/user-attachments/files/16507039/otel_demo.zip)

 [jaydeluca](https://github.com/jaydeluca) ,[laurit](https://github.com/laurit) Please find my spring application. attached for your reference.

**Reply by laurit** [reply, marked as answer]
Firstly, we provide 2 appenders for logback. One is for exporting logs via otlp and the other one is for for injecting trace and span id into log. You are using the wrong one, you need https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/logback/logback-mdc-1.0/library. Secondly trace/span id is injected based on the span that is current in the context. You need to change your code that creates the span to
```java
        Span span = tracer.spanBuilder("my-span").startSpan();
        try (Scope scope = span.makeCurrent()) {
            logger.info("Hello world");
            // Your business logic
            return "Hello, World!";
        } finally {
            span.end();
        }
```
after that you should see
```
2024-08-07 16:39:15 INFO  com.example.otel_demo.MyController [0242afd085e6513ab9171cdb1d888b68:93d76694944f2110] - Hello world
```
in the output.

**Comment by Yogeshsuch** [comment]
I refer https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/logback/logback-appender-1.0/library this as well
